libs/langchain/langchain/document_loaders/parsers/pdf.py

```diff
@@ -149,3 +149,56 @@ def lazy_parse(self, blob: Blob) -> Iterator[Document]:
                 )
                 for page in doc.pages
             ]
+
+
+class AmazonTextractPDFParser(BaseBlobParser):
+    """sends PDF files to  Amazon Textract and parses them to generate Documents.
+    multi-page documents have to reside on S3, single page PDF can be on local file system
+    multi-page documents that do not reside on S3 will throw an ValueError
+    """
+
+    try:
+        import textractcaller as tc
+    except ImportError:
+        raise ModuleNotFoundError(
+            "Could not import amazon-textract-caller python package. "
+            "Please install it with `pip install amazon-textract-caller`."
+        )
+
+    def __init__(self, textract_features:list[tc.Textract_Features]=[], boto3_textract_client=None) -> None:
```

Comment:
Would prefer to use a shorter name for the boto3 client.
```suggestion
    def __init__(self, textract_features:list[tc.Textract_Features]=[], boto3_textract_client=None) -> None:
```